workflow/Snakefile_training

```diff
@@ -0,0 +1,91 @@
+from snakemake.utils import min_version
+min_version("7.0")
+
+import pandas as pd
+import os
+import yaml
+
+
+configfile: "config/config_training.yaml"
+conda: "mamba"
+
+model_config = pd.read_table(config["model_config"], na_values="").fillna("None").set_index("model", drop=False)
+dataset_config = pd.read_table(config["dataset_config"], na_values="").set_index("biosample", drop=False)
+
+def make_paths_absolute(obj, base_path):
+	"""
+	Use absolute paths to be compatible with github submodules
+	Recursively go through the dictionary and convert relative paths to absolute paths.
+	"""
+	if isinstance(obj, dict):
+		for key, value in obj.items():
+			obj[key] = make_paths_absolute(value, base_path)
+	elif isinstance(obj, str):
+		# We assume all strings are paths. If converting the string
+		# to an absolute path results in a valid file, then the str was a path
+		new_file = os.path.join(base_path, obj)
+		if os.path.exists(new_file):
+			return new_file
+	return obj
+
+config = make_paths_absolute(config, os.getcwd())
+# Need to manually make results_dir an absolute path since above may
+# not work if results_dir folder isn't created
+# If results_dir is already an absolute path, this is a no-op
+config["results_dir"] = os.path.join(os.getcwd(), config["results_dir"])
+RESULTS_DIR = config["results_dir"]
+E2G_DIR_PATH = config["E2G_DIR_PATH"]
+
+def get_e2g_config(config):
```

Comment:
seems like there's a lot of shared code between this and the normal Snakefile. We can define shared functions into their own snakemake modules (utils.smk, arc_e2g.smk, etc) and just reference those

Reply:
I kept them separate because the file locations are slightly different between the two (eg. writing files to either a `dataset` output directory or `dataset/model` directory, and exactly how the different multiome files will be associated with different models/datasets is not clear yet (need to collab with Wei-Lin to confirm). And there are a few specific params that are unique to training/applying so not sure how useful this would be.